packages/commerce-sdk-react/src/provider.tsx

```diff
@@ -17,8 +20,33 @@ export interface CommerceAPIProviderProps {
     currency: string
 }
 
+export const CommerceAPIContext = React.createContext({} as ApiClients)
+
+// TODO: how to test? test in typescript template for now
 const CommerceAPIProvider = (props: CommerceAPIProviderProps): ReactElement => {
-    return <></>
+    const {children, clientId, organizationId, shortCode, siteId, proxy} = props
+
+    const config = {
+        proxy,
+        headers: {},
+        parameters: {
+            clientId,
+            organizationId,
+            shortCode,
+            siteId,
+        },
+        throwOnBadResponse: true,
+    }
+
+    // TODO: auth logic should use the helpers from commerce-sdk-isomorphic ?
+    const commerceAPI = new CommerceAPI(config)
```

Comment:
This `CommerceAPI` is the CommerceAPI class taken from the retail-react-app. The class does a few things, including:
- using `Proxy` to automatically inject the access token to the headers of API requests
- does manage the tokens and save them to either cookie or local storage.

Reply:
@kevinxh I saw the comment about `useShopperLoginHelpers` managing the tokens. How about if we delegate it to a class like this instead?